subiquity/models/timezone.py

```diff
@@ -0,0 +1,56 @@
+# Copyright 2021 Canonical, Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import logging
+
+log = logging.getLogger('subiquity.models.timezone')
+
+
+class TimeZoneModel(object):
+    """ Model representing timezone"""
+
+    timezone = ''
+
+    def __init__(self):
+        self._request = None
+        self.timezone = ''
```

Comment:
I don't understand the distinction between _request and timezone. Can you add a comment about this?

Reply:
Splitting it this way allows reporting `timezone: geoip` in /var/log/installer/autoinstall-user-data.  Users who want a specific timezone can set that instead, and it'll be persisted with their selected timezone.
Adding code comment to match - I can easily see myself forgetting the above in a few months.